lib/identity_cache/query_api.rb

```diff
@@ -450,6 +450,11 @@ def prefetch_one_association(association, records)
         end
         next_level_records
       end
+
+      def expire_primary_index(id) # :nodoc:
+        return unless primary_cache_index_enabled
+        IdentityCache.cache.delete(rails_cache_key(id))
```

Comment:
```suggestion
        id = type_for_attribute(primary_key).cast(id)
        IdentityCache.cache.delete(rails_cache_key(id))
```

might make sense with this being a public method to protect against mis-use (e.g. passing in a string id)